### PR TITLE
Automatically check and generate `Everything.agda`s

### PR DESCRIPTION
**Pull request by m-yac**
This PR adds:
- `make check-everythings`, which lists any files which are not somehow imported in some `Everything.agda`, with the exception of files in Cubical/Experiments
- `make gen-everythings`, which generates `Everything.agda` files in all directories except Core, Foundations, Codata, and Experiments

When I actually ran `make check-everythings` I got the following output:
```
> make check-everythings
runhaskell ./GenEverythings.hs check Experiments
Found some files which are not imported in any Everything.agda:
 Cubical.Foundations.HAEquiv
 Cubical.Foundations.Equiv.Fiberwise
 Cubical.Foundations.Univalence.Universe
 Cubical.HITs.Pushout.Flattening
 Cubical.HITs.Ints.QuoInt.Properties
 Cubical.Data.DescendingList
 Cubical.Data.SumFin
 Cubical.Data.Maybe
 Cubical.Data.Nat.Literals
make: *** [check-everythings] Error 1
```
This is indeed correct, so this PR also adds these files to their respective `Everything.agda`'s to ensure `make test` passes.

One problem with #26 is that it depended on the package `FileManip`. This Haskell script also depends on a package (there's no way to do something like this without it) but it is `directory`, which is a [dependency of agda](https://github.com/agda/agda/blob/d4a7e0e562bd8f1c66784f46d859a6c4d8ab445e/Agda.cabal#L196) and a part of the [Haskell Platform](https://www.haskell.org/platform/contents.html). So hopefully anyone trying to build this library will have this package installed.

One thing I didn't do in this PR is actually run `make gen-everythings`, because I'm not sure if it gives the output folks want. It alphabetizes the imports and neither opens nor exports any of them. Take a look [here](https://github.com/m-yac/cubical/compare/genEverythings...m-yac:genEverythings-expr). If this is what's expected then I'll merge this branch in.

### CODE REVIEW

**Comment by m-yac** [comment]
I also added `make check-README`, which checks whether every `Cubical.X.Everything` file is imported in `Cubical.README`. Not sure if we want this on by default, so I didn't include it in `make test`. It's current output is below:
```
> make check-README
runhaskell ./GenEverythings.hs checkREADME
Found some Everything.agda's which are not imported in README.agda:
 Cubical.Structures.Everything
 Cubical.Modalities.Everything
 Cubical.ZCohomology.Everything
 Cubical.Induction.Everything
 Cubical.Categories.Everything
make: *** [check-README] Error 1
```

**Comment by mortberg** [comment]
Thanks a lot!

Can this be combined with Travis somehow? It would be nice if Travis would complain if someone forgot an import.

Once this is merged I guess we'll have a PR where we delete all Everything files except the ones in Core, Foundations, Codata, and Experiments and then have `gen-everythings` run every time one runs `make`? I guess we should not have automatically generated files in the repo.

**Comment by m-yac** [comment]
> Can this be combined with Travis somehow? It would be nice if Travis would complain if someone forgot an import.

Already done, `make test` currently calls `check-everythings`.

> Once this is merged I guess we'll have a PR where we delete all Everything files except the ones in Core, Foundations, Codata, and Experiments and then have `gen-everythings` run every time one runs `make`? I guess we should not have automatically generated files in the repo.

Hmm, well what is the purpose of the `Everything` files? This comment made me realize I don't actually know. Are they just for `make check`?

**Comment by mortberg** [comment]
> > Once this is merged I guess we'll have a PR where we delete all Everything files except the ones in Core, Foundations, Codata, and Experiments and then have `gen-everythings` run every time one runs `make`? I guess we should not have automatically generated files in the repo.
> 
> Hmm, well what is the purpose of the `Everything` files? This comment made me realize I don't actually know. Are they just for `make check`?

Not only, it's very convenient to have them when making larger changes to the library. If I rename some lemmas and want to check that all files in a package still typecheck I just load the corresponding Everything file and emacs will take me to the place something fails.

File: Cubical/Modalities/Lex.agda [hunk 1]
```diff
@@ -143,8 +143,8 @@ module Combinators where
       B : A → Type ℓ′
       C : Σ A B → Type ℓ′′
 
-  λ/coe⟨_⟩_ : (p : A ≡ A′) → ((x : A′) → B (coe1→0 (λ i → p i) x)) → ((x : A) → B x)
-  λ/coe⟨_⟩_ {B = B} p f = coe1→0 (λ i → (x : p i) → B (coei→0 (λ j → p j) i x)) f
+  λ/coe⟨_⟩_ : (p : A ≡ A′) → ((x : A′) → B (transport⁻ p x)) → ((x : A) → B x)
+  λ/coe⟨_⟩_ {B = B} p f = transport⁻ (λ i → (x : p i) → B (transp (λ j → p (i ∧ ~ j)) (~ i) x)) f
```
[the remark this answers is not in the document]
Comment (mortberg) [reply Cubical/Modalities/Lex.agda:147]: Ah, that was not what I meant with moving it to Experiments... I think using Cartesian Kan ops is a good thing that maybe should be encouraged

**Comment by mortberg** [comment]
> > Once this is merged I guess we'll have a PR where we delete all Everything files except the ones in Core, Foundations, Codata, and Experiments and then have gen-everythings run every time one runs make? I guess we should not have automatically generated files in the repo.
> 
> Doing this would break the `README.agda` file. I think it also might be nice to not have to regenerate these files every time when pulling from the repo.
> 
> Speaking of `README.agda`, it currently doesn't include every new module (see `check-README`). Is this a problem?

I forgot there was a README.agda file... It seems to be quite out of date. Maybe we can just remove it?

I think it is a good thing to always generate the Everything files when building the library, that way we're sure that they're up to date and that everything will be checked.

File: GNUmakefile [hunk 2]
```diff
@@ -1,12 +1,13 @@
 AGDA_EXEC=agda
 RTS_OPTIONS=+RTS -H3G -RTS
 AGDA=$(AGDA_EXEC) $(RTS_OPTIONS)
+EVERYTHINGS=runhaskell ./Everythings.hs
 
 .PHONY : all
 all : check
```
[the remark this answers is not in the document]
Comment (mortberg) [reply GNUmakefile:7]: Yeah, that would be nice

File: Cubical/Modalities/Lex.agda (hunk 1 above)
[the remark this answers is not in the document]
Comment (mortberg) [reply Cubical/Modalities/Lex.agda:147]: Maybe they should have stayed in Foundations actually

**Comment by m-yac** [comment]
> I forgot there was a README.agda file... It seems to be quite out of date. Maybe we can just remove it?

I think it's there for `make listings`, see #216. Perhaps we could switch to @phadej's original suggestion if we no longer want this file?

> I think it is a good thing to always generate the Everything files when building the library, that way we're sure that they're up to date and that everything will be checked.

Great, so all that's left is to add `gen-everythings` to `make`, and ensure the Everything files in every directory except `Core`, `Foundations`, `Codata`, and `Experiments`are deleted and `.gitignore`ed. I get the impression you want this in a separate PR once this one is merged?

**Comment by mortberg** [comment]
> I think it's there for `make listings`, see #216. Perhaps we could switch to @phadej's original suggestion if we no longer want this file?
> 

I don't really care as I've never built the html documentation myself. It seems pointless to have a README file if it's out of date and it's something that at least I will quickly forget about updating. Maybe make an issue and we can discuss what to do later?

> > I think it is a good thing to always generate the Everything files when building the library, that way we're sure that they're up to date and that everything will be checked.
> 
> Great, so all that's left is to add `gen-everythings` to `make`, and ensure the Everything files in every directory except `Core`, `Foundations`, `Codata`, and `Experiments`are deleted and `.gitignore`ed. I get the impression you want this in a separate PR once this one is merged?

This doesn't sound too complicated, so let's just do it all in this one then.

**Comment by m-yac** [comment]
> This doesn't sound too complicated, so let's just do it all in this one then.

Done.

I also made another change: having `make check` use the module list from the README.agda file instead of using a hard-coded list (see my changes to the makefile and README.agda).

The idea is that the new workflow for someone adding a new top-level directory is that they add its Everything file to README.agda and include a short description. If they add a new top-level directory and don't do the former, `make test` will fail. I think this is a better solution to our current workflow of just having to remember to add a line to the makefile.

**Comment by mortberg** [comment]
It seems like you forgot to remove the Everything files that will be automatically generated?

**Pull request merged**